lib/src/query/query.dart

```diff
@@ -0,0 +1,563 @@
+    return _opWithEqual(p, ConditionOp.gt, caseSensitive, withEqual);
+  }
+
+  QueryCondition lessThan(String p, {bool caseSensitive = false, bool withEqual = false}) {
+    return _opWithEqual(p, ConditionOp.lt, caseSensitive, withEqual);
+  }
+
+  QueryCondition operator == (String p) => equals(p);
+//  QueryCondition operator != (String p) => notEqual(p); // not overloadable
+}
+
+class QueryIntegerProperty extends QueryProperty {
+  QueryIntegerProperty(int entityId, int propertyId, int type) : super(entityId, propertyId, type);
+
+  QueryCondition _op(int p, ConditionOp cop) {
+    final c = IntegerCondition(cop, type, p, 0);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition _opList(List<int> list, ConditionOp cop) {
+    final c = IntegerCondition.fromList(cop, type, list);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition equals(int p) {
+    return _op(p, ConditionOp.eq);
+  }
+
+  QueryCondition notEqual(int p) {
+    return _op(p, ConditionOp.not_eq);
+  }
+
+  QueryCondition greaterThan(int p) {
+    return _op(p, ConditionOp.gt);
+  }
+
+  QueryCondition lessThan(int p) {
+    return _op(p, ConditionOp.lt);
+  }
+
+  QueryCondition operator < (int p) => lessThan(p);
+  QueryCondition operator > (int p) => greaterThan(p);
+
+  QueryCondition inside(List<int> list) {
+    return _opList(list, ConditionOp.inside);
+  }
+
+  QueryCondition notInList(List<int> list) {
+    return _opList(list, ConditionOp.not_in);
+  }
+
+  QueryCondition notIn(List<int> list) {
+    return notInList(list);
+  }
+
+  // QueryCondition operator != (int p) => notEqual(p); // not overloadable
+  QueryCondition operator == (int p) => equals(p);
+}
+
+class QueryDoubleProperty extends QueryProperty {
+
+  QueryDoubleProperty(int entityId, int propertyId, int type) : super(entityId, propertyId, type);
+
+  QueryCondition _op(ConditionOp op, double p1, double p2) {
+    final c = DoubleCondition(op, type, p1, p2);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition between(double p1, double p2) {
+    return _op(ConditionOp.tween, p1, p2);
+  }
+
+  // TODO determine default tolerance: between (target - tolerance, target + tolerance)
+  QueryCondition equals(double p, {double tolerance = 0.01}) {
+    final absTolerance = tolerance.abs();
+    return between(p - absTolerance, p + absTolerance);
+  }
+
+  QueryCondition greaterThan(double p) {
+    return _op(ConditionOp.gt, p, null);
+  }
+
+  QueryCondition lessThan(double p) {
+    return _op(ConditionOp.lt, p, null);
+  }
+
+  QueryCondition operator < (double p) => lessThan(p);
+  QueryCondition operator > (double p) => greaterThan(p);
+  QueryCondition operator == (double p) => equals(p);
+}
+
+class QueryBooleanProperty extends QueryProperty {
+  QueryBooleanProperty(int entityId, int propertyId) : super(entityId, propertyId, type);
```

Comment:
```suggestion
  QueryBooleanProperty(int entityId, int propertyId, int type) : super(entityId, propertyId, type);
```

Reply:
Sorry, about that oversight. I also found it.